photon-ml/src/main/scala/com/linkedin/photon/ml/data/KeyValueScore.scala

```diff
@@ -77,16 +82,20 @@ protected[ml] class KeyValueScore(val scores: RDD[(Long, Double)]) extends RDDLi
     val subtractedScores =
       this.scores
         .fullOuterJoin(that.scores)
-        .mapValues { case (thisScore, thatScore) => thisScore.getOrElse(0.0) - thatScore.getOrElse(0.0) }
+        .mapValues { case (thisScore, thatScore) =>
+          require(!thisScore.isEmpty || !thatScore.isEmpty, s"Should not have reached this state. A full outer join " +
```

Comment:
Can this case actually happen? If so, it would be worth explaining the circumstances in a comment. If not, we should retire this `require`.